File: 0167-two-sum-ii---input-array-is-sorted/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) 
    {
        int lo, hi;
        lo = 0; hi = numbers.size()-1;
        
        vector<int> result;
        while(lo < hi)
        {
            if(numbers[lo] + numbers[hi] == target)
            {
                result.push_back(lo + 1);
                result.push_back(hi + 1);
                return result;
            }
            else if(numbers[lo] + numbers[hi] < target)
            {
                lo++;
            }
            else
            {
                hi--;
            }
        }
        
        return vector<int>();
    }
};

File: 0167-two-sum-ii---input-array-is-sorted/solution.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        lo, hi = 0, len(numbers)-1
        
        while lo < hi:
            if numbers[lo] + numbers[hi] < target:
                lo += 1
            elif numbers[lo] + numbers[hi] > target:
                hi -= 1
            else:
                return [lo+1, hi+1]

        return []
